### PR TITLE
Fix snapshot bug.

### PR DESCRIPTION
**Pull request by ry**
Emailed V8 people about it: https://groups.google.com/d/topic/v8-users/JpNdd8OfnmU/discussion

### CODE REVIEW

File: deno2/deno.cc [hunk 1]
```diff
@@ -251,7 +251,7 @@ v8::StartupData MakeSnapshot(v8::StartupData* prev_natives_blob,
   }
 
   auto snapshot_blob =
-      creator->CreateBlob(v8::SnapshotCreator::FunctionCodeHandling::kKeep);
+      creator->CreateBlob(v8::SnapshotCreator::FunctionCodeHandling::kClear);
```
Comment (piscisaureus) [inline deno2/deno.cc]: I would suggest to add a comment here, noting that (maybe) kKeep would be better if it didn't make v8 crash.

Looking at mksnapshot, it uses:
* `kClear` in `CreateSnapshotDataBlob()`
* `kKeep` in `WarmUpSnapshotDataBlob()`

Comment (ry) [reply deno2/deno.cc]: Comment added. I believe kKeep is meant for usage with WarmUps. Let's see what V8 says about it.

**Comment by seishun** [comment]
After this commit, I'm getting a segfault when building deno2:

```
~/deno/deno2$ ./tools/build.py --use_ccache --debug
ninja -C /home/drkoljan/deno/deno2/out/Debug deno
ninja: Entering directory `/home/drkoljan/deno/deno2/out/Debug'
[1/5] ACTION //:create_snapshot_deno(//build/toolchain/linux:clang_x64)
FAILED: gen/natives_deno.cc gen/snapshot_deno.cc 
python ../../v8/tools/run.py /home/drkoljan/deno/deno2/out/Debug/snapshot_creator gen/bundle/main.js natives_blob.bin snapshot_blob.bin gen/natives_deno.cc gen/snapshot_deno.cc


#
# Fatal error in ../../v8/src/objects/shared-function-info-inl.h, line 593
# Debug check failed: outer_scope_info()->IsScopeInfo() || is_toplevel().
#
#
#
#FailureMessage Object: 0x7ffec5dafc10
==== C stack trace ===============================

    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x1361fc3]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x136154b]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x135a328]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x135a095]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x468be7]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x467da7]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x44f89f]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator() [0x44cac3]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5) [0x7fa087360b45]
    /home/drkoljan/deno/deno2/out/Debug/snapshot_creator(_start+0x2a) [0x44c02a]
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "./tools/build.py", line 67, in <module>
    main()
  File "./tools/build.py", line 44, in main
    run(["ninja", "-C", buildDir, TARGET])
  File "./tools/build.py", line 50, in run
    subprocess.check_call(args, env=env)
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ninja', '-C', '/home/drkoljan/deno/deno2/out/Debug', 'deno']' returned non-zero exit status 1
```

**Comment by ry** [comment]
@seishun Thanks - yea we see it too and are working on it.

**Pull request merged**